drafts/25-xxx-specifications.txt

```diff
@@ -555,16 +555,59 @@ nd20. The result of evaluating a processor-dependent directive is processor-depe
 4 Macro recognition and expansion
 ---------------------------------
 
+This section is currently incomplete.  A self-contained specification
+for macro recognition and expansion will appear in a forthcoming
+paper.
+
+4.1 Comparison to Macro recognition and expansion in CPP
+--------------------------------------------------------
+
+We intend for macro recognition and expansion to operate almost
+exactly as specified in C 2023 section 6.10.5, with the following
+minor exceptions:
+
+me10. When determining argument boundaries in the invocation of a
+      function-like macro, FPP ignores commas surrounded by matching
+      sets of `[  ]` or `(/  /)` brackets, in addition to matching
+      sets of `(  )` parentheses (as specified in C 2023 section
+      6.10.5-11).
+
+me30. As specified by C 2023 6.10.5-10: "Within the sequence of
+      preprocessing tokens making up an invocation of a function-like
+      macro, new-line is considered a normal white-space character."
+
+me31. In addition, within the sequence of preprocessing tokens making
+      up an invocation of a function-like macro, a `&` character
+      immediately followed by a newline is immediately replaced
+      by a single blank character before subsequent processing.
+
+me32. The combination of rules me30 and me31 above imply that
+      when the invocation of a function-like macro is broken across
+      Fortran source lines, no line continuation character is
+      required, but that a Fortran &-style line continuation character
+      may optionally be used without changing the expanded result.
+
+me35. When performing macro recognition on a Fortran source line, if
+      the sequence of preprocessing tokens making up an invocation of
+      a function-like macro includes a `!` token, the rest of the
+      current line is treated as a comment; specifically, all tokens
+      starting with that `!` up through and including the next
+      subsequent newline are immediately deleted and replaced with a
+      single blank character, before further processing of the macro
+      arguments.
```

Comment:
Let's talk about this one. I'm sure you've thought about this in more detail than I have tonight, but I am not sure how I get a `!` token in a replacement list that is wasn't subject to interpretation as a C `!` operator.